### ISSUE TITLE
random reads from ds18b20

### ISSUE THREAD

**Issue by serein7** [OpenAgInitiative/openag_ds18b20#1]
I'm not sure is this a problem of my particular ds18b20 unit or the code, so just creating this issue to leave a trace in case of somebody also will have the similar problem.

In my case, with original driver's code, sensor reads random values like this:
>1473843656.342426,water_temperature,90.17658233642578
1473843649.736343,water_temperature,52.426631927490234
1473843643.133209,water_temperature,53.20883560180664
1473843636.525459,water_temperature,91.30783081054688
1473843629.923698,water_temperature,55.18354034423828
1473843623.316691,water_temperature,53.83393859863281
1473843616.716038,water_temperature,96.67256164550781
1473843610.105907,water_temperature,72.19131469726562
1473843603.503399,water_temperature,108.9841537475586
1473843596.895801,water_temperature,95.73313903808594
1473843590.29251,water_temperature,66.7006607055664
1473843583.684945,water_temperature,68.33355712890625
1473843577.061365,water_temperature,-1.106060266494751
1473843570.467081,water_temperature,0.6882352828979492
1473843563.868621,water_temperature,101.69747924804688
1473843557.264649,water_temperature,83.28856658935547

Only when I change constructor statment from `_sensors.setWaitForConversion(false);` to `_sensors.setWaitForConversion(true);` it starts to work as expected.

**Comment by LeonChambers** [comment]
Doing `_sensors.setWaitForConversion(true)` results in blocking reads that will hurt performance when running with multiple sensors. My only guess for why this happens is that when only 1 sensor is in use, the Arduino requests a temperature from the sensor and then checks if the temperature value is ready before the sensor has even processed the initial request. The sensor says that the new value is ready because a value was ready before the request, but then the value is actually junk. If this is the problem, then the fix would be to enforce a minimum interval between the call to `requestTemperatures` and the call to `getTempC`.

**Comment by serein7** [comment]
Yes, I do have only one sensor attached on OneWire bus. Thanks for the tip, I'll try a slight delay and see if it helps.

**Comment by gordonbrander** [comment]
If this is the case, we should add a note to docs about this footgun.

**Comment by LeonChambers** [comment]
I think we should just fix the code so that this doesn't happen. It actually shouldn't be that hard

**Comment by serein7** [comment]
Yes, it is possible to detect number of sensors attached and add delay. However, I was only able to add blocking delay(300) to get it working, which is bad. Wherever I was trying to emulate delay with 'millis() - _ query_last _time_' it was not working .. so I did not commit anything yet.
